llm/memory.go

```diff
@@ -437,3 +448,33 @@ func projectorMemoryRequirements(filename string) (weights, graphSize uint64) {
 
 	return weights, graphSize
 }
+
+// estimateKvCacheSize determines the memory required for K or V cache based on the quantization type
+func estimateKvCacheSize(cacheType string, numCtx, blockCount, embeddingHeadCount, headCountKV uint64, isEmbeddingModel bool) uint64 {
```

Comment:
Instead of passing in `isEmbeddingModel` here, could we force the cacheType to `f16` here: https://github.com/ollama/ollama/pull/6279/files#diff-c5f3086b077dd2de3fd6639b7e326e0e8dd0273483486d131ec54911d8503fccR129

Reply:
Oh whoops missed this comment earlier, taking a look now

Reply:
Oh yes, that would be nicer, will submit an update shortly

Reply:
Updated.

Reply:
Added it back in - that broke the tests, It's late here now and I'm getting tired / making silly mistakes so might have to have a look at this again tomorrow if I get a chance.

Reply:
Right, I've had my 6 hours of beauty sleep and am thinking clearly again 🤣 ,

I started to implement this but discovered it might not be the best approach. Here's why:

The `isEmbeddingModel` flag serves two distinct purposes:

1. In server.go/validation.go, it validates user input to ensure only f16/f32 cache types are allowed for embedding models
2. In estimateKvCacheSize, it ensures accurate memory calculations by forcing non-f32 cache types to f16

While we could try to force the cache type earlier, we need the memory estimation logic to "know" about embedding models to properly calculate memory requirements. If we removed the flag, we'd either have to duplicate the embedding model detection logic or potentially get inaccurate memory estimates.

I've kept the flag but improved the code by:
- Moving the validation logic to ValidateKVCacheType
- Simplifying the cache type handling in server.go
- Making the embedding model handling more explicit in estimateKvCacheSize

Let me know if I've misunderstood or if you see a better way we need to handle this!

Reply:
Got it, thanks. Sorry for the confusion on my side as well, agreed we should move it to one file (see comment bellow).

Reply:
No worries at all, it's a complex thing and we're operating across other ends of timezones 🤣.

Also - my Golang is not fantastic yet, so it has required some of your direction - thanks for that, I know it's additional work on top of anything else you have going on.